### PR TITLE
[MI-100] Prevent calls to the API when rendering the grid if unauthenticated.

### PR DESCRIPTION
**Pull request by joseconsador**
[MI-100]

Save authenticated flag to the database.
Do not display Create Ticket and Dashboard/Grid if the extension is not setup properly.

Steps to reproduce:
1. Login to the magento backend.
2. Go to Zendesk > Configuration.
3. In the General tab, make all fields empty except for Zendesk Domain.
    In the Admin Backend tab, set "Show support tickets on admin dashboard" to "Yes".
4. Click Save.
1. Go to Zendesk > Dashboard

Expected Result:
1. "Couldn't authenticate you" should appear as an error message.
2. On step #5 - the configuration notice should be displayed

Actual Result:
1. A php warning (attached) appears, along with the standard error message.

screen shot 2015-06-22 at 3 48 57 pm
1. On step #5 - a fatal error is thrown and halts the process (attached)

screen shot 2015-06-22 at 4 00 41 pm

/cc @miogalang @jwswj @mmolina @atroche 
### References
- Jira link: https://zendesk.atlassian.net/browse/MI-100
### Risks
- Medium - Extension functionality may be affected.
## QA notes
- Developer mode must be enabled

### CODE REVIEW

File: src/app/code/community/Zendesk/Zendesk/etc/config.xml [hunk 1]
```diff
@@ -19,7 +19,7 @@
 <config>
     <modules>
         <Zendesk_Zendesk>
-            <version>2.0.6</version>
+            <version>2.0.7</version>
         </Zendesk_Zendesk>
```
Comment (miogalang) [inline src/app/code/community/Zendesk/Zendesk/etc/config.xml:23]: Can you put the bumping in a new commit

**Comment by miogalang** [comment]
:+1:

**Comment by jwswj** [comment]
Can't we just check the domain name, username and password/token fields, rather than introducing a new key-value?

**Comment by joseconsador** [comment]
@jwswj Those values could be incorrect, which will cause the extension to make some API calls that returns unexpected responses.

**Comment by jwswj** [comment]
I'm on :-1: on this until we find an approach which doesn't involve saving an attribute based on successful auth. The reason being Magento can't always know about auth. What about is the token gets changed from inside Zendesk, this error would still happen. Or if someone changes their subdomain and forgets to update the Magento settings.

File: src/app/code/community/Zendesk/Zendesk/controllers/Adminhtml/ZendeskController.php [hunk 2]
```diff
@@ -691,8 +691,9 @@ protected function getMassActionResponse($response, $ids, $message = '%d out of
 
     private function _domainConfigured()
     {
-        $domain = Mage::getStoreConfig('zendesk/general/domain');
-        if(!$domain) {
+        $authenticated = Mage::getStoreConfig('zendesk/general/authenticated');
```
Comment (joseconsador) [inline src/app/code/community/Zendesk/Zendesk/controllers/Adminhtml/ZendeskController.php:694]: This line causes an error. :-1:

**Pull request closed without merging**